pkg/blobstore/grpcservers/byte_stream_server.go

```diff
@@ -37,27 +39,49 @@ func (s *byteStreamServer) Read(in *bytestream.ReadRequest, out bytestream.ByteS
 	if err != nil {
 		return err
 	}
-	if compressor != remoteexecution.Compressor_IDENTITY {
-		return status.Error(codes.Unimplemented, "This service does not support downloading compressed files")
-	}
-
-	r := s.blobAccess.Get(out.Context(), digest).ToChunkReader(in.ReadOffset, s.readChunkSize)
-	defer r.Close()
+	switch compressor {
+	case remoteexecution.Compressor_IDENTITY:
+		r := s.blobAccess.Get(out.Context(), digest).ToChunkReader(in.ReadOffset, s.readChunkSize)
+		defer r.Close()
 
-	for {
-		readBuf, readErr := r.Read()
-		if readErr == io.EOF {
-			return nil
-		}
-		if readErr != nil {
-			return readErr
+		for {
+			readBuf, readErr := r.Read()
+			if readErr == io.EOF {
+				return nil
+			}
+			if readErr != nil {
+				return readErr
+			}
+			if writeErr := out.Send(&bytestream.ReadResponse{Data: readBuf}); writeErr != nil {
+				return writeErr
+			}
 		}
-		if writeErr := out.Send(&bytestream.ReadResponse{Data: readBuf}); writeErr != nil {
-			return writeErr
+
+	case remoteexecution.Compressor_ZSTD:
+		b := s.blobAccess.Get(out.Context(), digest)
+		zstdWriter, err := zstd.NewWriter(&readStreamWriter{out: out})
```

Comment:
But for reading and writing, should we make sure to call `zstd.WithDecoderConcurrency()` / `zstd.WithEncoderConcurrency()`? We already have concurrency at the request level. It's probably not a good idea to also let this spawn `GOMAXPROCS` goroutines.